src/server.rs

```diff
@@ -469,4 +473,112 @@ pub extern "C" fn rustls_server_session_get_sni_hostname(
         }
         rustls_result::Ok
     }
-}
\ No newline at end of file
+}
+
+/// The TLS Client Hello information provided to a ClientHelloCallback function.
+/// `sni_name` is the SNI servername provided by the client (not 0 terminated) with
+/// `sni_name_len` as its length. If the client did not provide an SNI, the name
+/// length is 0.
+#[allow(non_camel_case_types)]
+#[repr(C)]
+pub struct rustls_client_hello {
+    sni_name: *const c_char,
+    sni_name_len: size_t,
+    signature_schemes: *const c_ushort,
+    signature_schemes_len: size_t,
+}
+
+/// Any context information the callback will receive when invoked.
+type ClientHelloUserData = *mut c_void;
+
+/// Called when the TLS ClientHello message was received and successfully decoded.
+type ClientHelloCallback = unsafe extern "C" fn(
```

Comment:
These typedefs wind up getting exported in the .h, so let's use C-style naming for them: `rustls_client_hello_user_data` and `rustls_client_hello_callback` (this is something I missed in my own branch; I'll update it to follow my own advice :-D)